nncf/onnx/graph/model_transformer.py

```diff
@@ -68,166 +99,101 @@ def transform(self, transformation_layout: TransformationLayout) -> onnx.ModelPr
-                                                               outputs=[*extra_model_outputs,
-                                                                        *model_outputs])
-        self._model = model_with_intermediate_outputs
-
-    def _get_extra_model_outputs(self,
-                                 nncf_graph: NNCFGraph,
-                                 onnx_graph: ONNXGraph,
-                                 transformations: List[ONNXOutputInsertionCommand]) -> Set[str]:
-        """
-        Collects extra model outputs based on transformations
-
-        :param nncf_graph: NNCFGraph
-        :param onnx_graph: ONNXGraph
-        :param transformations: lisf of the ONNXOutputInsertionCommand
-        :return: list of the output names
-        """
-        extra_model_outputs = set()
-        input_edge_names = []
-
+        extra_model_outputs = []
         for transformation in transformations:
+            port_id = transformation.target_point.port_id
             node_name = transformation.target_point.target_node_name
-            if NNCFGraphNodeType.INPUT_NODE in node_name:
-                nncf_node_name = nncf_graph.get_node_by_name(transformation.target_point.target_node_name)
-                onnx_nodes_after_input_node = [edge.to_node for edge in nncf_graph.get_output_edges(nncf_node_name)]
-                for onnx_node_name in onnx_nodes_after_input_node:
-                    input_edge_names.append(onnx_graph.get_node_edge_names(onnx_node_name.node_name)['input'][0])
-                extra_model_outputs.update(input_edge_names)
-                input_edge_names = []
-            else:
-                if transformation.target_point.type == TargetType.POST_LAYER_OPERATION:
-                    edge_name = onnx_graph.get_node_edge_names(node_name)['output'][
-                        transformation.target_point.port_id]
-                elif transformation.target_point.type == TargetType.PRE_LAYER_OPERATION:
-                    edge_name = onnx_graph.get_node_edge_names(node_name)['input'][
-                        transformation.target_point.port_id]
-                else:
-                    raise RuntimeError
-                extra_model_outputs.add(edge_name)
-            extra_model_outputs.update(input_edge_names)
-        return extra_model_outputs
-
-    def _insert_outputs(self, model: onnx.ModelProto, outputs: List[str] = None) -> onnx.ModelProto:
+            transform_type = transformation.target_point.type
+            target_edge_name = self._get_pre_post_layer_target_edge(port_id, node_name, transform_type, onnx_graph)
+            if target_edge_name is None:
+                raise RuntimeError('Could not find the edge corresponding to node {}'.format(node_name))
+            extra_model_outputs.append(target_edge_name)
+        return ONNXModelTransformer._insert_outputs(self._model, outputs=[*extra_model_outputs, *model_outputs])
```

Comment:
Deepcopy missed?

Reply:
this transformation returns a new model instance, thus no demanding deepcopy